[BUG] Change Filament feature uses heater and sensor for E2 when option E1 is selected

### Configuration Files

Test on latest bugfix (downloaded today):

[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/5543394/Marlin.zip)

Only the most minimal changes needed to get Marlin to run on my printer with LCD and buttons. Features enabled are setup for dual extruder and sensors.

### Steps to Reproduce

Sanity check sensors:

```
#define WATCH_TEMP_PERIOD 1

Unplug E2 heater (E2 temp sensor must stay connected or an alarm will sound when the printer boots).

LCD:
  Temperature > Nozzle 1: > Preheat End 1

Alarm sounds: "Heating Failed: E1" -- OK!

Reset.

LCD:
  Temperature > Nozzle 2: > Preheat End 2

Alarm sounds: "Heating Failed: E2" -- OK!

Reset.
```

ACTUAL TEST:

```
LCD:
  Change Filament > Change Filament E2 > Preheat

Alarm sounds: "Heating Failed: E2" -- OK!

Reset.

LCD:
  Change Filament > Change Filament E1 > Preheat <--- Heating for E1!

(Displays temperature of E2 sensor)

Alarm sounds: "Heating Failed: E2" <--- This is the bug! Uses E2 heater/sensor!

Reset.
```

**Expected behavior:**

Filament options for working with E1 affect E1.

**Actual behavior:**

Filament options for working with E1 affect E2!

Note that filament options for working with E2 correctly affect E2 (this is not a problem with configuration, pins, or swapped cables). All options for pre-heating as well as interacting with E1 and E2 affect the correct tool when connected with serial USB with Repetier.

I get a compile error when attempting to use dummy sensor `998`:

```
In file included from Marlin\src\HAL\AVR\../../inc/MarlinConfig.h:49:0,
                 from Marlin\src\HAL\AVR\HAL.cpp:24:
Marlin\src\HAL\AVR\../../inc/SanityCheck.h:1642:4: error: #error "TEMP_0_PIN (required for TEMP_SENSOR_0) not defined for this board."
   #error "TEMP_0_PIN (required for TEMP_SENSOR_0) not defined for this board."
    ^
```

--- a/Marlin/src/lcd/menu/menu_filament.cpp
+++ b/Marlin/src/lcd/menu/menu_filament.cpp
@@ -91,7 +91,7 @@ void _menu_temp_filament_op(const PauseMode mode, const int8_t extruder) {
   BACK_ITEM(MSG_BACK);
   #if PREHEAT_COUNT
     LOOP_L_N(m, PREHEAT_COUNT)
-      ACTION_ITEM_N_S(m, ui.get_preheat_label(m), MSG_PREHEAT_M, _change_filament_with_preset);
+      ACTION_ITEM_N_S(extruder, ui.get_preheat_label(m), MSG_PREHEAT_M, _change_filament_with_preset);
   #endif
   EDIT_ITEM_FAST_N(int3, extruder, MSG_PREHEAT_CUSTOM, &thermalManager.temp_hotend[extruder].target,
     EXTRUDE_MINTEMP, thermalManager.heater_maxtemp[extruder] - HOTEND_OVERSHOOT,

